### PR TITLE
LB-1174 : Hides the empty years in all time listening chart

### PR DESCRIPTION
**Pull request by pranavkonidena**
# Problem
This hides the empty years in all time listening chart. Earlier, even though there was no listening data for some years they were shown, now only one empty year is shown (as discussed in IRC)
https://tickets.metabrainz.org/browse/LB-1174

# Solution
Just changed the function to keep the years with non zero listens and if listening data for only one year is present, only that year and the prev year will be shown
Screenshot 2023-12-06 at 4 18 24 PM

# Action

### CODE REVIEW

**Comment by pranavkonidena** [comment]
Hi, could someone pls review if this looks fine and let me know?
Thanks

File: frontend/js/src/stats/UserListeningActivity.tsx [hunk 1]
```diff
@@ -335,22 +335,31 @@ export default class UserListeningActivity extends React.Component<
...
-
       totalYears += 1;
       if (yearData) {
         const date = new Date(yearData.from_ts * 1000);
-        allTimeData.push({
-          id: date.toLocaleString("en-us", dateFormat),
-          thisRangeCount: yearData.listen_count,
-          thisRangeTs: yearData.from_ts,
-        });
+        if (yearData.listen_count > 0) {
```
Comment (amCap1712) [inline frontend/js/src/stats/UserListeningActivity.tsx:347]: One suggestion, don't remove all years with 0 listen count but only the ones before any listening activity is encountered. Say a user started submitting data in 2017, didn't submit anything in 2018 but then started again in 2019. The graph should start from 2017 and 2018 should be shown in the graph. With the current checks, 2018 would be hidden.

**Comment by pranavkonidena** [comment]
Screenshot 2023-12-07 at 4 24 14 PM
This looks fine but in the case the user has only one year of data, it doesn't look too good.. can I do one empty year and rest data as suggested by @MonkeyDo in IRC @amCap1712 
This is what it will look like if only one year of data is present
Screenshot 2023-12-07 at 4 26 24 PM

File: frontend/js/src/stats/UserListeningActivity.tsx (hunk 1 above)
Comment (pranavkonidena) [reply frontend/js/src/stats/UserListeningActivity.tsx]: Hey, any update on this?

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
Looks pretty good to me with the latest changes!
I have a small comment on variable naming* and you might have to resolve some merge conflicts (we reorganized files in another PR merged today) but otherwise it's ready to merge.

*"There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors."

**Review by MonkeyDo — APPROVED** [review]

**Comment by MonkeyDo** [comment]
I hope you don't mind, it was sucha  small change I took the liberty to do it myself and get the PR merged :)

**Pull request merged**